scheduler/src/cook/mesos/util.clj

```diff
@@ -24,44 +24,58 @@
             [metatransaction.core :refer (db)]
             [metrics.timers :as timers]
             [plumbing.core :as pc :refer (map-vals map-keys)])
-  (:import [java.util Date]))
+  (:import [java.util Date]
+           [com.google.common.cache CacheBuilder]
+           [java.util.concurrent TimeUnit]))
+
+
+(defn new-cache []
+  "Build a new cache"
+  (-> (CacheBuilder/newBuilder)
+      (.maximumSize 1000000)
+      ;; if its not been accessed in 2 hours, whatever is going on, its not being visted by the
+      ;; scheduler loop anymore. E.g., its probably failed/done and won't be needed. So,
+      ;; lets kick it out to keep cache small.
```

Comment:
Maybe we should put this info in the docstring?

Reply:
As discussed, we'll revisit how to thread these (and the resulting caches) in a followup JIRA. Dependency injection, globals, or something similar.

Reply:
What I meant was that maybe the info in the comment should live in the docstring instead

Reply:
My prior is that it will need a lot less time than 2 hours. Can we measure when last access was?

Reply:
@dposada Its an implementation detail, I don't see why we should expose it in a docstring.
@wyegelwel Probalby fine. I chose 2 hours to be uncontroversial. Do you have a specific timeout we should use for expiratoin? We could try to measure it, but its going to be workload and configuration dependent, so we'd want to measure in production.  IMO, not worth deep diving that far.

Reply:
Did we get @wyegelwel's concern resolved?

Reply:
Sorry, missed Scott's comment. If it is easy to measure, I think it is worth measuring. If not, then as long as we are measuring the overall size than we can probably punt on the 2 hours unless the cache is very large.

Reply:
We have an open JIRA to add metrics collection from the caches -- #739.

Reply:
Ok, good enough for me. I think #739 should get merged before we cut a release with this feature (or have it defaulted off)